### PR TITLE
Adding requirements docs for CloudStack provider

### PR DESCRIPTION
**Pull request by chrisnegus**
*Issue #:* [#3658](https://github.com/aws/eks-anywhere/issues/3658) 

*Description of changes:*
This PR creates the requirements page for creating EKS Anywhere clusters on a CloudStack provider.

Feel free to comment. It's still on hold as I work through a couple of issues.

### CODE REVIEW

File: docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md [hunk 1]
```diff
@@ -5,3 +5,63 @@ weight: 15
...
+### Prepare Administrative machine
+Set up an Administrative machine as described in [Install EKS Anywhere ]({{< relref "../../getting-started/install/" >}}).
+
+### Prepare a CloudStack environment
+
+To prepare a CloudStack environment to run EKS Anywhere, you need the following:
+
+* A CloudStack 4.14 or later environment. CloudStack 4.16 is used for examples in these docs.
+* Capacity to deploy 6-10 VMs.
+* One network in CloudStack to use for the cluster. EKS Anywhere clusters need access to CloudStack through the network to enable self-managing and storage capabilities.
```
Comment (vignesh-goutham) [inline docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md:20]:
```suggestion
* One shared network in CloudStack to use for the cluster. EKS Anywhere clusters need access to CloudStack through the network to enable self-managing and storage capabilities.
```

File: docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md [hunk 2]
```diff
@@ -5,3 +5,63 @@ weight: 15
...
+* A Red Hat Enterprise Linux ISO image built into a qcow2 image as described in [artifacts]({{< relref "../artifacts/" >}}).
```
Comment (vignesh-goutham) [inline docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md:21]:
```suggestion
* A Red Hat Enterprise Linux qcow2 image built using image-builder tool as described in [artifacts]({{< relref "../artifacts/" >}}).
```

File: docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md [hunk 3]
```diff
@@ -5,3 +5,63 @@ weight: 15
...
+* User credentials to create VMs and attach networks in CloudStack.
```
Comment (vignesh-goutham) [inline docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md:22]: A cloudstack user with access and secret key ?

File: docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md [hunk 4]
```diff
@@ -5,3 +5,63 @@ weight: 15
...
+    * Account (account): Account with permission to create a cluster (optional, admin by default).
+    * Credentials (credentialsRef): Credentials provided in an ini file used to access the CloudStack API endpoint. See [CloudStack Getting started]({{< relref "../../getting-started/production-environment/" >}}) for details.
+    * Domain (domain):  The CloudStack domain in which to deploy the cluster (optional, ROOT by default)
+    * Management endpoint (managementApiEndpoint): Endpoint for a cloudstack client to make API calls to client. 
+    * Zone network (zone.network): Either name or ID of the network. 
+* CloudStack machine configuration: For each set of machines (for example, you could configure separate set of machines for control plane, worker, and etcd nodes), obtain the following information. This must be predefined in the cloudStack instance and identified by name or ID:
+    * Compute offering (computeOffering): Choose an existing compute offering (such as m4-large), reflecting the amount of resources to apply to each VM.
```
Comment (vignesh-goutham) [inline docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md:65]: Can we use something like `large-instance` as example and not m4-large? m4-large might not necessarily translate well for cloudstack users.

File: docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md (hunk 3 above)
Comment (vignesh-goutham) [reply docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md:22]: We should also need to document the list of Cloudstack permissions required for the user.

Comment (mrog) [reply docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md:22]: > A cloudstack user with access and secret key ?

The user needs to have an API Key and a Secret Key.

File: docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md (hunk 1 above)
Comment (chrisnegus) [reply docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md:20]: Done.

File: docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md (hunk 2 above)
Comment (chrisnegus) [reply docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md:21]: Done.

File: docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md (hunk 3 above)
Comment (chrisnegus) [reply docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md:22]: Okay. I noted the CloudStack API key and Secret key.

File: docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md (hunk 4 above)
Comment (chrisnegus) [reply docs/content/en/docs/reference/cloudstack/cloudstack-prereq.md:65]: Done.

**Review by vignesh-goutham — APPROVED** [review]
/approve

**Pull request merged**